src/bp-members/classes/class-bp-members-notices-rest-controller.php

```diff
@@ -0,0 +1,1090 @@
+		 * @param WP_REST_Response  $retval  The response data.
+		 * @param WP_REST_Request   $request The request sent to the API.
+		 */
+		do_action( 'bp_members_notices_rest_get_item', $notice, $response, $request );
+
+		return $response;
+	}
+
+	/**
+	 * Check if a given request is allowed to get a notice.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 * @return WP_Error|bool
+	 */
+	public function get_item_permissions_check( $request ) {
+		$error  = new WP_Error(
+			'bp_rest_authorization_required',
+			__( 'Sorry, you are not allowed to see this notice.', 'buddypress' ),
+			array(
+				'status' => rest_authorization_required_code(),
+			)
+		);
+		$retval = $error;
+
+		if ( is_user_logged_in() ) {
+			$notice = bp_members_get_notice( $request->get_param( 'id' ) );
+			if ( empty( $notice->id ) ) {
+				$retval = new WP_Error(
+					'bp_rest_invalid_id',
+					__( 'Sorry, this notice does not exist.', 'buddypress' ),
+					array(
+						'status' => 404,
+					)
+				);
+			} elseif ( bp_current_user_can( 'bp_moderate' ) ) {
+					$retval = true;
+			} else {
+				$retval = $error;
+
+				if ( isset( $notice->target, $notice->priority ) && 127 !== (int) $notice->priority ) {
+					if ( ( 'contributors' === $notice->target && bp_current_user_can( 'edit_posts'  ) ) || 'community' === $notice->target ) {
+						$retval = true;
+					}
+				}
+			}
+		}
+
+		/**
+		 * Filter the  notices `get_item` permissions check.
+		 *
+		 * @since 15.0.0
+		 *
+		 * @param bool|WP_Error   $retval  Returned value.
+		 * @param WP_REST_Request $request The request sent to the API.
+		 */
+		return apply_filters( 'bp_members_notices_rest_get_item_permissions_check', $retval, $request );
+	}
+
+	/**
+	 * Create a notice.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function create_item( $request ) {
+		// Setting context.
+		$request->set_param( 'context', 'edit' );
+
+		$notice_id = bp_members_save_notice(
+			array(
+				'title'    => $request->get_param( 'title' ),
+				'content'  => $request->get_param( 'content' ),
+				'target'   => $request->get_param( 'target' ),
+				'priority' => $request->get_param( 'priority' ),
+				'date'     => $request->get_param( 'date' ),
+				'url'      => $request->get_param( 'action_url' ),
+				'text'     => $request->get_param( 'action_text' ),
+			)
+		);
+
+		if ( is_wp_error( $notice_id ) ) {
+			$notice_id->add_data(
+				array(
+					'status' => 500,
+				)
+			);
+
+			return $notice_id;
+		}
+
+		// The notice we just created will be active.
+		$notice        = bp_members_get_notice( $id );
+		$fields_update = $this->update_additional_fields_for_object( $notice, $request );
+
+		if ( is_wp_error( $fields_update ) ) {
+			return $fields_update;
+		}
+
+		$retval   = $this->prepare_item_for_response( $notice, $request );
```

Comment:
This method already sets `rest_ensure_response`. 🤔

Reply:
Do you know why we set it again in line 349?

Reply:
I have no idea to be honest, I haven't touched this part from the BP REST plugin version 🤔

Reply:
I checked all other BP REST API endpoints as well as the `WP_REST_Posts_Controller`: using the `rest_ensure_response()` twice (inside the `prepare_item_for_response()` & into the request callback) is also practiced. I don't know why, but it looks to be the way to do it... So let's carry on using it twice! As this [function](https://developer.wordpress.org/reference/functions/rest_ensure_response/) is only making sure a `WP_REST_Response` object is returned, it's hurting noone: twice is always better than never!